jobs/wf2-5-createEvents.js

```diff
@@ -4,68 +4,70 @@ fn(state => {
 });
 
 // Fetch TEI's for each patient
-each('encounters[*]', state => {
-  const encounter = state.data;
-
-  return get(
+each(
+  'encounters[*]',
+  get(
     'trackedEntityInstances',
-    {
+    state => ({
       ou: 'l22DQq4iV3G',
-      filter: [`jGNhqEeXy2L:Eq:${encounter.patient.uuid}`],
-    },
+      filter: [`jGNhqEeXy2L:Eq:${state.data.patient.uuid}`],
+    }),
     {},
     state => {
-      // console.log(
-      //   JSON.stringify(
-      //     state.data.trackedEntityInstances[0].trackedEntityInstance
-      //   )
-      // );
       state.TEIs[encounter.patient.uuid] =
         state.data.trackedEntityInstances[0].trackedEntityInstance;
 
       return state;
     }
-  )(state);
-});
+  )
+);
 
-// Create events fore each encounter
-each('encounters[*]', async state => {
-  const { oclMappings, data, TEIs } = state;
+// Prepare DHIS2 data model for create events
+fn(state => {
+  const { oclMappings, TEIs } = state;
 
-  const encounterDate = data.encounterDatetime.replace('+0000', '');
+  const encountersMapping = state.encounters.map(data => {
+    const encounterDate = data.encounterDatetime.replace('+0000', '');
 
-  const pluckObs = arg => data.obs.filter(ob => ob.concept.uuid === arg)[0];
-  const pluckOcl = arg =>
-    oclMappings.filter(ocl => ocl.from_concept_name_resolved === arg)[0];
+    const pluckObs = arg => data.obs.find(ob => ob.concept.uuid === arg);
+    const pluckOcl = arg =>
+      oclMappings.find(ocl => ocl.from_concept_name_resolved === arg);
 
-  const obs1 = pluckObs('da33d74e-33b3-495a-9d7c-aa00a-aa0160');
-  const obs2 = pluckObs('da33d74e-33b3-495a-9d7c-aa00a-aa0177');
+    const obs1 = pluckObs('da33d74e-33b3-495a-9d7c-aa00a-aa0160');
+    const obs2 = pluckObs('da33d74e-33b3-495a-9d7c-aa00a-aa0177');
 
-  const oclMap1 = obs1 && pluckOcl(obs1.value.display);
-  const oclMap2 = obs2 && pluckOcl(obs2.value.display);
+    const oclMap1 = obs1 && pluckOcl(obs1.value.display);
+    const oclMap2 = obs2 && pluckOcl(obs2.value.display);
 
-  const valueForEncounter1 = oclMap1 ? oclMap1.to_concept_name_resolved : '';
-  const valueForEncounter2 = oclMap2 ? oclMap2.to_concept_name_resolved : '';
+    const valueForEncounter1 = oclMap1 ? oclMap1.to_concept_name_resolved : '';
+    const valueForEncounter2 = oclMap2 ? oclMap2.to_concept_name_resolved : '';
```

Comment:
This is much less important, but a few things about this bother me.

What do you think about using this pattern instead? This is untested and I'm not sure the variable names are right...
```
    const conceptIds = [
      'da33d74e-33b3-495a-9d7c-aa00a-aa0160',
      'da33d74e-33b3-495a-9d7c-aa00a-aa0177',
    ];
    const [valueForEncounter1, valueForEncounter2] = conceptIds.map(id => {
      const obs = data.obs.find(ob => ob.concept.uuid === id);
      if (obs) {
        const ocl = oclMappings.find(
          ocl => ocl.from_concept_name_resolved === obs.value.display
        );
        if (ocl) {
          return ocl.to_concept_name_resolved;
        }
      }
      return '';
    });
    ```

Reply:
Actually don't worry about this - when we improve the ocl adaptor we can come back and revisit this algorithm.